lib/oli_web/live/new_course/new_course.ex

```diff
@@ -432,13 +434,16 @@ defmodule OliWeb.Delivery.NewCourse do
     end
   end
 
-  def handle_info({:section_created, section_slug}, socket) do
-    socket = put_flash(socket, :info, "Section successfully created.")
+  def handle_info({:section_created, section_id, section_slug}, socket) do
+    Task.Supervisor.start_child(Oli.TaskSupervisor, fn ->
+      depot_desc = SectionResourceDepot.depot_desc()
+      DistributedDepotCoordinator.init_if_necessary(depot_desc, section_id, SectionResourceDepot)
+    end)
 
-    {:noreply,
-     redirect(socket,
-       to: Routes.live_path(OliWeb.Endpoint, OliWeb.Sections.OverviewView, section_slug)
-     )}
+    socket
+    |> put_flash(:info, "Section successfully created.")
+    |> redirect(to: ~p"/sections/#{section_slug}/manage")
+    |> noreply()
```

Comment:
Nitpick: helper functions can sometimes make code look "prettier," they aren't always necessary, especially when the standard approach is already straightforward and widely understood. The current implementation of the `{:no_reply, socket}` follows the standard pattern, which is often preferable for consistency and maintainability
WDYT?

Reply:
It makes sense what you are saying. What you’re saying makes sense. What I’ve found is that we have a combination of two patterns in our codebase:

Type 1:

```
{:noreply,
  assign(socket,
  key_1: :val_1,
  key_2: :val_2,
  ...: ...
  key_N: :val_N,
  )}
```

Type 2:

```
socket = assign(socket,
  key_1: :val_1,
  key_2: :val_2,
  ...: ...
  key_N: :val_N,
  )

{:noreply,, socket}
```

There’s nothing inherently wrong with these approaches, but there are trade-offs. Type 1 is widely used in our codebase. However, for me, it’s not immediately obvious where the second tuple ends, which creates a lot of pain points when working with this pattern. It’s not very reader-friendly. While writing this pattern might feel straightforward, reading it can be challenging. As writers, we have context because we’re starting the sentence, but readers don’t always have the same advantage.

On the other hand, Type 2 is more structured and easier to follow. But it’s sometimes not the preferred choice among developers, perhaps because we approach writing and reading code differently.

The approach I’ve taken here is inspired by similar patterns I’ve seen in other projects. Additionally, [German Velazco](https://www.elixirstreams.com/tips/awesome-ok-noreply-liveview-helpers?utm_source=elixir-merge) has expressed support for this style.

My intention in adding this function is not just to make the code “prettier” but to make it easier to read. Given that we frequently use the Type 1 approach in our project, using `noreply/1` plays well with the pipe operator and doesn’t introduce significant overhead when applied.

Please let me know what you think.